apps/omg_watcher_info/lib/omg_watcher_info/api/transaction.ex

```diff
@@ -77,19 +107,53 @@ defmodule OMG.WatcherInfo.API.Transaction do
   Given order finds spender's inputs sufficient to perform a payment.
   If also provided with receiver's address, creates and encodes a transaction.
   """
-  @spec create(UtxoSelection.order_t()) :: UtxoSelection.advice_t()
+  @spec create(order_t()) :: create_t()
   def create(order) do
-    utxos = DB.TxOutput.get_sorted_grouped_utxos(order.owner)
-    UtxoSelection.create_advice(utxos, order)
+    with {:ok, inputs} <-
+           order.owner
+           |> DB.TxOutput.get_sorted_grouped_utxos()
+           |> select_inputs(order) do
+      create_transaction(inputs, order)
+    end
+  end
+
+  # Given an `order`, finds spender's inputs sufficient to perform a payment.
+  # If also provided with receiver's address, creates and encodes a transaction.
+  @spec select_inputs(utxos_map_t(), order_t()) :: inputs_t()
+  defp select_inputs(utxos, %{payments: payments, fee: fee}) do
+    token_utxo_selection =
+      payments
+      |> UtxoSelection.needed_funds(fee)
+      |> UtxoSelection.select_utxo(utxos)
+
+    with {:ok, funds} <- UtxoSelection.funds_sufficient?(token_utxo_selection) do
+      utxo_count =
+        funds
+        |> Stream.map(fn {_, utxos} -> length(utxos) end)
+        |> Enum.sum()
+
+      case utxo_count do
+        n when n <= Transaction.Payment.max_inputs() ->
+          stealth_merged_utxos =
+            utxos
+            |> UtxoSelection.prioritize_merge_utxos(funds)
+            |> UtxoSelection.add_utxos_for_stealth_merge(Map.new(funds))
+
+          {:ok, stealth_merged_utxos}
+
+        _ ->
+          {:error, :too_many_inputs}
+      end
+    end
   end
 
   @spec include_typed_data(UtxoSelection.advice_t()) :: UtxoSelection.advice_t()
   def include_typed_data({:error, _} = err), do: err
 
-  def include_typed_data({:ok, %{transactions: txs} = advice}),
+  def include_typed_data({:ok, txs}),
```

Comment:
you're mixing public and private functions here. it also seems the function needs to be written in multiline